changes/serializers.py

```diff
@@ -47,14 +47,14 @@ class AdminChangeSerializer(serializers.Serializer):
 
 class ReceiveWhatsAppEventSerializer(serializers.Serializer):
 
-    class HookSerializer(serializers.Serializer):
-        event = serializers.ChoiceField(['message.direct_outbound.status'])
-    hook = HookSerializer()
+    class StatusSerializer(serializers.Serializer):
 
-    class EventSerializer(serializers.Serializer):
-        status = serializers.ChoiceField(['unsent'])
+        class ErrorSerializer(serializers.Serializer):
+            code = serializers.CharField(required=False)
+            title = serializers.CharField(required=True)
 
-        class MessageMetadata(serializers.Serializer):
-            junebug_message_id = serializers.UUIDField()
-        message_metadata = MessageMetadata()
-    data = EventSerializer()
+        id = serializers.CharField(required=True)
+        status = serializers.ChoiceField(['failed'])
```

Comment:
I think we should have a test to confirm this, because I'm not 100% sure how this will work, but what if in the list of statuses that we're given, some of them are `failed`, and some of them are `succeeded`? Will the entire request fail with a 400, or will some of the statuses go through, and some of them fail? I'm not sure how this works with ListFields.

Also, if it does go through because some of them match, we probably need to filter it in the view to only action on the failed ones (unless the serializer does this for us?).

Either way, I think we should have a test to confirm that this works the way we expect it to.

Reply:
Good catch, I'll write a test to see what happens and handle accordingly.